staking/app/StakeConnection.ts

```diff
@@ -541,8 +559,7 @@ export class StakeConnection {
       throw Error(`Unexpected account state ${vestingAccountState}`);
```

Comment:
```
    if (
      vestingAccountState !=
        VestingAccountState.UnvestedTokensPartiallyLocked &&
      vestingAccountState != VestingAccountState.UnvestedTokensFullyUnlocked
    ) {
      throw Error(`Unexpected account state ${vestingAccountState}`);
```
These checks are only relevant for `lockAllUnvested`, you should move them up there